clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -1136,24 +1162,38 @@ private PartitionRecords parseCompletedFetch(CompletedFetch completedFetch) {
                     log.trace("Updating last stable offset for partition {} to {}", tp, partition.lastStableOffset);
                     subscriptions.updateLastStableOffset(tp, partition.lastStableOffset);
                 }
+
+                if (partition.preferredReadReplica.isPresent()) {
+                    long expireTimeMs = time.milliseconds() + metadata.metadataExpireMs();
```

Comment:
The call to system time here is a bit vexing. Would be nice if we only had to call this when we actually needed to update the read replica.

Reply:
Sure, I can push it down into SubscriptionState

Reply:
Ended up deferring the time call with a supplier